ui/imports/shared/views/AssetsView.qml

```diff
@@ -62,42 +69,78 @@ ColumnLayout {
             settingsKey: "WalletAssets"
         }
 
-        readonly property bool hasCommunityAssets: communityAssetsLV.count
-
         function hideAllCommunityTokens(communityId) {
-            const tokenSymbols = ModelUtils.getAll(communityAssetsLV.model, "symbol", "communityId", communityId)
+            const tokenSymbols = ModelUtils.getAll(regularAssetsLV.model, "symbol", "communityId", communityId)
             d.controller.settingsHideCommunityTokens(communityId, tokenSymbols)
         }
-    }
 
-    component CustomSFPM: SortFilterProxyModel {
-        id: customFilter
-        property bool isCommunity
+        readonly property SubmodelProxyModel assetsWithFilteredBalances: SubmodelProxyModel {
+            id: assetsWithFilteredBalances
+            sourceModel: root.assets
+            submodelRoleName: "balances"
+            delegateModel: SortFilterProxyModel {
+                sourceModel: submodel
+                filters: ExpressionFilter {
+                    expression: {
+                        root.networkFilters
+                        root.addressFilters
+                        return root.networkFilters.split(":").includes(chainId+"") &&
+                                (!!root.addressFilters ? root.addressFilters.toUpperCase() === account.toUpperCase() : true)
+                    }
+                }
+            }
+        }
 
-        sourceModel: root.assets
+        function getTotalBalance(balances, decimals) {
```

Comment:
Ugh really do we want to sum the numbers by using JS on the client side?

Reply:
I know, its not ideal but we tried to use the SumAggregator here but it wont work, this works fairly quickly, until we have a better solution for this.

Reply:
I see, probably not a big deal as the amount of assets is kinda limited/finite :)

Reply:
yes!